server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourcePhase.java

```diff
@@ -46,7 +46,7 @@ public StoredFieldsSpec storedFieldsSpec() {
             public void process(HitContext hitContext) {
                 String index = fetchContext.getIndexName();
                 if (fetchContext.getSearchExecutionContext().isSourceEnabled() == false) {
-                    if (fetchSourceContext.hasFilter()) {
+                    if (fetchSourceContext.hasFilter() || fetchSourceContext.includeVectors() == Boolean.TRUE || fetchSourceContext.includeVectors() == Boolean.FALSE) {
```

Comment:
It's OK to only check `fetchSourceContext.hasFilter()` here. This is one of those cases where the `hasFilter` call answering the question "Are there user-supplied filters?". Since we filter vectors by default, they are not necessarily user-defined.

Reply:
I agree but don't we want to send the exception when they specify a `true` or `false` value from the request when the source is not enabled?

Reply:
Not in my opinion. I interpret `includeVectors` as "include/exclude embeddings in _source, if they exist". The key phrase in that interpretation is "if they exist". The user is depending on us to do the right thing without them needing to think too much about it. In the case of disabled _source, the right thing in my mind is to treat `includeVectors` as a no-op.

I could be convinced to throw the exception if `includeVectors` is `true`, depending on the wider group's interpretation of the setting. But throwing this exception on `false` makes no sense to me in any interpretation.

@carlosdelest @kderusso WDYT?

Reply:
I agree that throwing on false should be a no-op, but I think if `includeVectors` is explicitly set to `true` in this case we should throw because it's impossible to do this without enabling `_source`.

Reply:
Ok, consensus reached. Throw on `includeVectors == true`.